affirm_example/__init__.py

```diff
@@ -104,6 +104,86 @@ def read_charge(charge_id):
+        template_data["{0}_url".format(charge_action)] = url_for(".admin_do",
+                                                                 charge_action=charge_action,
+                                                                 charge_id=template_data["charge_id"],
+                                                                 **kwargs)
+
+    return flask.render_template("user_confirm.html", **template_data)
+
+# Affirm Transactions REST API
+def create_transaction(checkout_token):
+    create_transaction_url = "{0}/transactions".format(app.config["AFFIRM"]["TRANSACTIONS_API_URL"])
+    print create_transaction_url
+    request_args = {}
+    return requests.post(create_transaction_url,
+                         data=json.dumps({
+                             "transaction_id": checkout_token
+                         }),
+                         headers={"Content-Type": "application/json"},
+                         auth=(app.config["AFFIRM"]["PUBLIC_API_KEY"],
+                               app.config["AFFIRM"]["SECRET_API_KEY"]),
+                         **_get_extra_request_args()).json()
+
+def read_transaction(transaction_id):
+    get_transaction_url = "{0}/transactions/{1}".format(app.config["AFFIRM"]["TRANSACTIONS_API_URL"], transaction_id)
+    print get_transaction_url
+    return requests.get(get_transaction_url,
+                        auth=(app.config["AFFIRM"]["PUBLIC_API_KEY"],
+                              app.config["AFFIRM"]["SECRET_API_KEY"]),
+                        **_get_extra_request_args()).json()
+
+def void_transaction(transaction_id):
+    void_transaction_url = "{0}/transactions/{1}/void".format(app.config["AFFIRM"]["TRANSACTIONS_API_URL"], transaction_id)
+    print void_transaction_url
+    return requests.post(void_transaction_url,
+                         auth=(app.config["AFFIRM"]["PUBLIC_API_KEY"],
+                               app.config["AFFIRM"]["SECRET_API_KEY"]),
+                         **_get_extra_request_args()).json()
+
+
+def refund_transaction(transaction_id, amount=10):
+    refund_transaction_url = "{0}/transactions/{1}/refund".format(app.config["AFFIRM"]["TRANSACTIONS_API_URL"], transaction_id)
+    print refund_transaction_url
+    return requests.post(refund_transaction_url,
+                         data=json.dumps({
+                             "amount": amount
+                         }),
+                         headers={"Content-Type": "application/json"},
+                         auth=(app.config["AFFIRM"]["PUBLIC_API_KEY"],
+                               app.config["AFFIRM"]["SECRET_API_KEY"]),
+                         **_get_extra_request_args()).json()
+
+def capture_transaction(transaction_id, amount=None):
+    capture_transaction_url = "{0}/transactions/{1}/capture".format(app.config["AFFIRM"]["TRANSACTIONS_API_URL"], transaction_id)
+    print capture_transaction_url
```

Comment:
Any reason to `print` the urls? I'm fine with it, but if there's no reason it's probably cleaner to leave it out.

Reply:
Not sure, I was following the pattern of existing functions in that file. It *was* helpful for debugging.

Reply:
Oh, my bad. Didn't realize other functions also had it. That sounds good to me!